src/ActionMenu.tsx

```diff
@@ -0,0 +1,132 @@
+import {List, ListPropsBase, GroupedListProps} from './ActionList/List'
+import {Item, ItemProps} from './ActionList/Item'
+import {Divider} from './ActionList/Divider'
+import Button, {ButtonProps} from './Button'
+import React, {useCallback, useRef, useState} from 'react'
+import Overlay from './Overlay'
+import {useFocusTrap} from './hooks/useFocusTrap'
+import {useFocusZone} from './hooks/useFocusZone'
+import {useAnchoredPosition} from './hooks/useAnchoredPosition'
+import {uniqueId} from './utils/uniqueId'
+
+export interface ActionMenuProps extends Partial<Omit<GroupedListProps, keyof ListPropsBase>>, ListPropsBase {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  renderAnchor?: (props: any) => JSX.Element
+  anchorContent?: React.ReactNode
+  onAction?: (props: ItemProps) => void
+}
+
+const ActionMenuItem = (props: ItemProps) => <Item role="menuitem" {...props} />
+
+ActionMenuItem.displayName = 'ActionMenu.Item'
+
+const ActionMenuBase = ({
+  anchorContent,
+  renderAnchor = <T extends ButtonProps>(props: T) => <Button {...props}>{anchorContent}</Button>,
+  renderItem = Item,
+  onAction,
+  ...listProps
+}: ActionMenuProps): JSX.Element => {
+  const anchorRef = useRef<HTMLElement>(null)
+  const overlayRef = useRef<HTMLDivElement>(null)
```

Comment:
Position likely isn't being applied properly on the overlay in this current state.  You would need to bring over `const [overlayRef, updateOverlayRef] = useRenderForcingRef<HTMLDivElement>()` from DropdownMenu.  Hopefully we circle back to Popover to make this simpler soon 😅